Sources/Routing/Routing/RouterNode.swift

```diff
@@ -9,7 +9,7 @@ final class RouterNode<Output> {
     var constants: [RouterNode<Output>]
 
     /// Parameter child node, if one exists.
-    var parameter: RouterNode<Output>?
+    var parameter: (node: RouterNode<Output>, type: LosslessDataConvertible.Type)?
```

Comment:
This PR opens us up to the possibility of registering both `"foo", Int.parameter` and `"foo", String.parameter`, so parameter could become an array of tuples if that's something we're interested in. I'd prefer to decide on that later down the road though because doing so would be non-trivial